internal/limitopen/limitopen.go

```diff
@@ -0,0 +1,101 @@
+package limitopen
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+
+	"github.com/reddit/baseplate.go/errorsbp"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+)
+
+// Open opens a path for read.
+//
+// It's similar to os.Open, but with the following differences:
+//
+// 1. It returns io.ReadCloser other than *os.File,
+//
+// 2. It returns the size reported by the system to the user.
+//
+// 3. The io.ReadCloser returned would be guaranteed to be not able to read
+//    beyond the size returned (for example, if you use this function to open
+//    /dev/zero, the system would return 0 as the size, and as a result when
+//    reading from r you would get EOF immediately, instead of being able to
+//    read from it indefinitely)
+//
+// It would never return both non-nil r and err.
+// When err is nil it's the caller's responsibility to close r returned.
+func Open(path string) (r io.ReadCloser, size int64, err error) {
+	var f *os.File
+	f, err = os.Open(path)
+	if err != nil {
+		return
+	}
```

Comment:
* Don't use bare returns, especially if it requires predeclaring variables
* Add context to errors, and use %w if you want them to be visible

```suggestion
	f, err := os.Open(path)
	if err != nil {
		return nil, 0, fmt.Errorf("opening file: %w", err)
	}
```